src/lib/phdi-building-blocks/phdi_building_blocks/schemas.py

```diff
@@ -0,0 +1,251 @@
+    """
+    Given a resource and a schema return a dictionary with values of the data
+    specified by the schema and associated keys defined by the variable name provided
+    by the schema.
+
+    :param resource: A FHIR resource on which to apply a schema.
+    :param schema: A schema specifying the desired values by FHIR resource type.
+    :return data: A dictionary containing the data extracted from the patient a long
+    with specified variable names.
+    """
+
+    data = {}
+    resource_schema = schema.get(resource.get("resourceType", ""))
+    if resource_schema is None:
+        return data
+    for field in resource_schema.keys():
+
+        path = resource_schema[field]["fhir_path"]
+        value = evaluate(resource, path)
+
+        if len(value) == 0:
+            data[resource_schema[field]["new_name"]] = ""
+        else:
+            selection_criteria = resource_schema[field]["selection_criteria"]
+            value = apply_selection_criteria(value, selection_criteria)
+            data[resource_schema[field]["new_name"]] = value
+
+    return data
+
+
+def make_resource_type_table(
+    resource_type: str,
+    schema: dict,
+    output_path: pathlib.PosixPath,
+    output_format: Literal["parquet"],
+    credential_manager: AzureFhirserverCredentialManager,
+):
+    """
+    Given a FHIR resource type, schema, and FHIR server credential manager create a
+    table containing the field from resource type specified in the the schema.
+
+    :param resource_type: A FHIR resource type.
+    :param schema: A schema specifying the desired values by FHIR resource type.
+    :param credential_manager: A credential manager for
+    a FHIR server.
+    """
+
+    output_path.mkdir(parents=True, exist_ok=True)
+    output_file_name = output_path / f"{resource_type}.{output_format}"
+
+    query = f"/{resource_type}"
+    response = query_fhir_server(credential_manager, query)
+
+    additional_page = True
+    writer = None
+    while additional_page:
+        if response.status_code != 200:
+            log_fhir_server_error(response.status_code)
+            break
+
+        # Load queried data.
+        query_result = json.loads(response.content)
+        raw_schema_data = []
+
+        # Extract values specified by schema from each resource.
+        for resource in query_result["entry"]:
+            values_from_resource = apply_schema_to_resource(
+                resource["resource"], schema
+            )
+            if values_from_resource != {}:
+                raw_schema_data.append(values_from_resource)
+
+        # Write data to parquet
+        writer = write_schema_table(
+            raw_schema_data, output_file_name, output_format, writer
+        )
+
+        # Check for an additional page of query results.
+        for link in query_result.get("link"):
+            if link.get("relation") == "next":
+                next_page_url = link.get("url")
```

Comment:
Change this variable name to `url` so that it can be reused in the `query_fhir_server(...)` call you make inside the `while` loop after the recommended change.